### ISSUE TITLE
Docs need a note on foreign key references to user

### ISSUE THREAD

**Issue by MichaelXavier** [Soostone/snaplet-persistent#14]
This is from a discussion between myself, @denibertovic, and @bitemyapp. Because of the way QQ scopes things, `schema.txt` in this repo is not in the same scope when the user includes the decls into their own app. The result is that persistent can't figure out that `SnapAuthUserId` is a primary key and thus if you reference it in another table, it will not create a foreign key. The solution is to manually tell persistent's TH that the column is a reference.

**Comment by MichaelXavier** [comment]
Working example: http://dpaste.com/35XTE73

**Comment by denibertovic** [comment]
The dpaste I shared with you has been set to expire (and apparently has) so I'm sharing
the code here:

    import qualified Snap.Snaplet.Auth.Backends.Persistent as SP

    share [mkPersist sqlSettings, mkMigrate "migrateAll"] $ [persistLowerCase|
      SomeStuffz
        userId SP.SnapAuthUserId Eq reference=snap_auth_user
        createdAt UTCTime default=CURRENT_TIMESTAMP
      deriving Eq Show Generic
    |] ++ SP.authEntityDefs

**Comment by bitemyapp** [comment]
@denibertovic you can make it highlight that as Haskell code in the following manner:

\`\`\`haskell
import Blah (myCode)
\`\`\`

Example:

```haskell
import Blah (myCode)
```

**Comment by denibertovic** [comment]
@bitemyapp Yeah I know. Just didn't think about it at the moment. Thanks, I've updated the snippet.

**Comment by bitemyapp** [comment]
@denibertovic cheers.

Also is it me or is Github missing a "view source" button for comments?

**Comment by denibertovic** [comment]
I don't remember seeing that so it might be just you. :P

**Comment by bitemyapp** [comment]
@denibertovic I mean it should be added. It's a turn of phrase.

**Comment by denibertovic** [comment]
Ah. Indeed. :)
